packages/devtools/lib/src/timeline/frames_bar_plotly.dart

```diff
@@ -305,3 +345,136 @@ class FramesBarPlotly {
     legendClick(domName, f);
   }
 }
+
+class SelectTrace {
+  SelectTrace(this.traceIndex, this.ptNumber, this.xValue, this.yValue);
+
+  final int traceIndex;
+  final int ptNumber;
+  final int xValue;
+  final num yValue;
+}
+
+class Selection {
+  Selection(this._domName, dynamic graphDiv)
+      : _data = getProperty(graphDiv, 'data');
+
+  final String _domName;
+  final List<Data> _data;
+  List<SelectTrace> selectInfo = [];
+
+  bool isSelected(List<int> xValues) =>
+      selectInfo.length == 2 &&
+      selectInfo[0].xValue == xValues[0] &&
+      selectInfo[1].xValue == xValues[1];
+
+  int get selectedPointNumber =>
+      selectInfo.isNotEmpty ? selectInfo[0].ptNumber : -1;
+
+  void select(List<int> pointNumbers, List<int> traces, List<int> xValues,
+      List<num> yValues) {
+    // Supports one bar selection and not selecting a currently selected bar.
+    assert(traces.length == 2 &&
+        traces[0] != FramesBarPlotly.gpuSelectTraceIndex &&
+        traces[1] != FramesBarPlotly.cpuSelectTraceIndex);
+
+    List<SelectTrace> oldSelectInfo = unselect();
+
+    // Maybe adjust our current pointNumbers (plotly term is an array index
+    // into data). If we messed with a trace and the old pointNumbers was before
+    // our new bar we need to adjust.
+    if (oldSelectInfo.isNotEmpty) {
+      int oldTrace0 = oldSelectInfo[0].traceIndex;
+      int oldPtNum0 = oldSelectInfo[0].ptNumber;
+      int oldTrace1 = oldSelectInfo[1].traceIndex;
+      int oldPtNum1 = oldSelectInfo[1].ptNumber;
+
+      // After unselecting we're selecting a point to the right of our
+      // old selection in a particular trace so we need to adjust to take into
+      // the old data we just restored in a trace. (Check both traces).
+      if (oldTrace0 == traces[0] && pointNumbers[0] >= oldPtNum0) {
+        pointNumbers[0] += 1;
+      }
+      if (oldTrace1 == traces[1] && pointNumbers[1] >= oldPtNum1) {
+        pointNumbers[1] += 1;
+      }
+    }
+
+    // Construct our new selection info.
+    int index = 0;
+    for (var trace in traces) {
+      selectInfo.add(SelectTrace(
+          trace, pointNumbers[index], xValues[index], yValues[index]));
+      index++;
+    }
+
+    // Make room for our selection bar remove the data we're selecting it will
+    // exist in the selection traces.
```

Comment:
nit: can you rephrase this comment or split into two sentences? I'm not sure what you mean here.